src/interfaces/IEngine.sol

```diff
@@ -57,6 +57,20 @@ interface IEngine {
         bytes[] conditions;
     }
 
+    /// @notice canExecute error codes
+    enum CanExecuteError {
```

Comment:
This is great. But maybe lets rename it not to use the word "Error" since `None` is not technically an error. Maybe "CanExecuteStatus" or "CanExecuteResponse"

Reply:
Also, it is returned when a user calls the `canExecute()` and in that case, it just represents a Response type even if it not `None`